ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -294,6 +294,8 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
           existingIdenticalEntries++;
           metrics.identicalPutAttemptCount.inc();
         } else {
+          logger.trace("COLLISION: For key {} in WriteSet with crc {}, index already has the key with journal crc {}",
+              info.getStoreKey(), info.getCrc(), index.journal.getCrcOfKey(info.getStoreKey()));
```

Comment:
(Correct me if am wrong) `COLLIDING` is because the key is not seen in journal (checked by `index.wasRecentlySeen(info)`). So, could `index.journal.getCrcOfKey(info.getStoreKey())` also be `null`?  (if so, which makes the log message ambiguous)

Reply:
We conservatively conclude that this is a collision case in the logic, when either of journal crc or incoming crc is null. The log simply prints that conclusion, along with what the values of the two crcs - and either or both can be null. As such I don't think this is ambiguous.